### PR TITLE
bridge address review comments

### PR DESCRIPTION
**Pull request by ancaantochi**
Address some review comments from a merged PR

### CODE REVIEW

File: mqtt/mqtt-bridge/src/controller.rs [hunk 1]
```diff
@@ -111,7 +114,7 @@ impl BridgeController {
                     }
 
                     if let Some(config) = self.bridge_handles.lock().await.get_mut(UPSTREAM) {
-                        if let Err(e) = config.send_update(&bridge_update).await {
+                        if let Err(e) = config.send_update(bridge_update.to_owned()).await {
```
Comment (ancaantochi) [inline mqtt/mqtt-bridge/src/controller.rs:117]: had to add to_owned here because it gets consumed

File: mqtt/mqtt-bridge/src/controller.rs [hunk 2]
```diff
@@ -101,7 +104,7 @@ impl BridgeController {
         };
 
         let updates = async move {
-            while let Some(update) = self.updates_receiver.recv().await {
```
Comment (and-rewsmith) [inline mqtt/mqtt-bridge/src/controller.rs:104]: recv() call was busy?

**Review by and-rewsmith — APPROVED** [review]

File: mqtt/mqtt-bridge/src/config_update.rs [hunk 3]
```diff
@@ -21,8 +22,10 @@ impl ConfigUpdater {
         }
     }
 
-    pub async fn send_update(&mut self, bridge_update: &BridgeUpdate) -> Result<(), Error> {
-        let diff = self.diff(&bridge_update);
+    pub async fn send_update(&mut self, bridge_update: BridgeUpdate) -> Result<(), Error> {
+        let diff = self.diff(bridge_update);
+
+        debug!("sending diff {:?}", diff.clone());
```
Comment (dmolokanov) [inline mqtt/mqtt-bridge/src/config_update.rs:28]:
```suggestion
        debug!("sending diff {:?}", diff);
```

File: mqtt/mqtt-bridge/src/controller.rs (hunk 1 above)
Comment (dmolokanov) [reply mqtt/mqtt-bridge/src/controller.rs:117]: which is fine? I guess we don't need `bridge_update` anywhere here
```suggestion
                        if let Err(e) = config.send_update(bridge_update).await {
```

File: mqtt/mqtt-bridge/src/config_update.rs (hunk 3 above)
Comment (dmolokanov) [reply mqtt/mqtt-bridge/src/config_update.rs:28]:
```rust
self.update(diff)
...
    fn update(&mut self, bridge_diff: BridgeDiff) {
        let (local_updates, remote_updates) = bridge_diff.into_parts();
        update_pump(local_updates, &mut self.current_forwards);

        update_pump(remote_updates, &mut self.current_subscriptions)
    }
...

fn update_pump(pump_diff: PumpDiff, current: &mut HashMap<String, TopicRule>) {
    let (added, removed) = pump_diff.into_parts();
    for rule in added {
        current.insert(rule.subscribe_to(), rule);
    }

    for rule in removed {
        current.remove(&rule.subscribe_to());
    }
}
....

fn diff_topic_rules(updated: Vec<TopicRule>, current: &HashMap<String, TopicRule>) -> PumpDiff 
...
 for sub in updated {
        if !current.contains_key(&sub.subscribe_to())
            || current
                .get(&sub.subscribe_to())
                .filter(|curr| *curr == &sub)
                == None
        {
            added.push(sub);
        }
    }
...

```

Comment (dmolokanov) [reply mqtt/mqtt-bridge/src/config_update.rs:28]: Notice how we removed cloning just changing value move semantic from borrowing.

**Review by dmolokanov — CHANGES_REQUESTED** [review]

File: mqtt/mqtt-bridge/src/controller.rs (hunk 1 above)
Comment (ancaantochi) [reply mqtt/mqtt-bridge/src/controller.rs]: Actually I noticed that to_owned is not needed, not sure why I thought was needed.

File: mqtt/mqtt-bridge/src/controller.rs (hunk 2 above)
Comment (ancaantochi) [reply mqtt/mqtt-bridge/src/controller.rs:104]: Denis suggested in the original PR to use next

**Review by dmolokanov — APPROVED** [review]
LGTM

**Pull request merged**